sound/soc/sof/sof-client.h

```diff
@@ -0,0 +1,85 @@
+/* SPDX-License-Identifier: (GPL-2.0-only) */
+
+#ifndef __SOUND_SOC_SOF_CLIENT_H
+#define __SOUND_SOC_SOF_CLIENT_H
+
+#include <linux/completion.h>
+#include <linux/debugfs.h>
+#include <linux/device.h>
+#include <linux/device/driver.h>
+#include <linux/kernel.h>
+#include <linux/list.h>
+#include <linux/virtual_bus.h>
+
+#define SOF_CLIENT_PROBE_TIMEOUT_MS 2000
+
+struct snd_sof_dev;
+
+enum sof_client_type {
+	SOF_CLIENT_AUDIO,
+	SOF_CLIENT_IPC,
```

Comment:
sorry if I missed it in an earlier revision. Not a reason for a new version unless you have to make one anyway. But in an incremental patch: `SOF_CLIENT_IPC` is for the IPC flood test, right? I think we should have the work "test" somewhere in the name. Otherwise it makes an impression that without this IPC isn't supported at all.

Reply:
agree with @lyakh  this should be SOF_CLIENT_IPC_TEST

Reply:
@lyakh no SOF_CLIENT_IPC is not specifically for IPC test. It is for everything thats not audio and the IPC_TEST_CLIENT is one of them. For example, the nodec client, probes client, ssp_dmic client will all be audio clients but ipc test client will not be. Does that make sense?

Reply:
I see `.type` set to `SOF_CLIENT_IPC` in the IPC flood test client, but I don't see that field used anywhere, so, it isn't quite clear to me what its meaning should be and how it should be used. If it is unused for now maybe remove it for now and decide later? Or you have a specific reasoning why you want a coarse granularity like that?

Reply:
ok I'll remove it for now